kmp-library/src/commonMain/kotlin/com/novoda/aws/v4/signer/CanonicalRequest.kt

```diff
@@ -0,0 +1,145 @@
+/*
+  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+  the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+
+  Copyright 2016 the original author or authors.
+ */
+package com.novoda.aws.v4.signer
+
+import com.novoda.aws.v4.signer.encoding.URLEncoding
+
+private const val S3_SERVICE = "s3"
+private const val QUERY_PARAMETER_SEPARATOR = '&'
+private const val QUERY_PARAMETER_VALUE_SEPARATOR = '='
+
+/**
+ * @author Richard Lucas
```

Comment:
So far we removed the author and license from the classes we migrated to kotlin. I'd vote being consisten here although I am not 100% sure, what's the way to go here.
Added an [issue](https://github.com/novoda/aws-v4-signer-java/issues/21) to reevaluate licensing before we do a release.